openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -91,13 +91,25 @@ def _get_custom_pacing_children(subsection, num_weeks):
     Return relative date items for the subsection and its children
     """
     items = [subsection]
+    all_problems_are_ora = True
+    has_empty_unit = True
     section_date_items = []
     while items:
         next_item = items.pop()
+        if next_item.category != 'sequential' and next_item.category != 'vertical':
+            has_empty_unit = False
         # Open response assessment problems have their own due dates
         if next_item.category != 'openassessment':
             section_date_items.append((next_item.location, {'due': timedelta(weeks=num_weeks)}))
             items.extend(next_item.get_children())
+            if next_item.category != 'sequential' and next_item.category != 'vertical':
```

Comment:
(nit rant, with no action on your part:)

I kind of hate whenever we add checks like this. And I'm guilty of it too, to be fair. But categories aren't as neat as we want. Consider that there are actually category types like `problemset` and `videosequence` (both of which act like `sequential`). Those are [slated for removal](https://openedx.atlassian.net/browse/DEPR-151). But just an example of how bad it is to have string comparisons like this strewn through our code. (And in that instance, very little of our code thinks to check for the aliases - and no one ever seemed to have thought to add a utility method for it like `sequence_types()` or `is_sequence()`?!)

Ideally, we'd have an xblock method like `xblock.is_hierarchy_block()` and that way any weird new custom xblock type could declare itself to be a hierarchy/structure block. Or an enumeration like we do for completion API where we say what style of xblock it is - `HIERARCHY`, or something else. Or even just a utility method like `is_hierarchy_block(xblock)` to at least keep the number of places we check this sort of thing to a minimum.

But the last time I looked at adding such a utility method myself, I was also struck by how very custom each of these checks really were. Like, often you want to know if it's a chapter or sequential, but not vertical. Or sequential and vertical, but not chapter. So I never tried to add any sanity myself. It just left me with a sense of dread anytime I see hardcoded category checks, because I know it's a mess we're leaving for future generations.

Reply:
Maybe having utility methods for each level? Or one where you can toggle each level on or off? Like:
```
is_chapter_type()
is_sequence_type()
is_vertical_type()
```
Or
```
is_hierarchy_type(chapter=True, sequence=True, vertical=True) or something smarter
```

Again, nothing you should do in this PR. Just daydreaming.